src/plugins/intel_gna/gna_plugin.cpp

```diff
@@ -1459,11 +1466,28 @@ GnaWaitStatus GNAPlugin::WaitFor(uint32_t request_idx, int64_t millisTimeout) {
                 fprintf(f, "\n\n");
             }
 #endif
-            ConvertToFloat(outputBlob->buffer(),
-                outputBlob->buffer(),
-                elementsPerBatch,
-                batchSize,
-                outputDesc.scale_factor);
+            switch (outputBlob->getTensorDesc().getPrecision()) {
+            case InferenceEngine::Precision::FP32 :
+                ConvertToFloat(outputBlob->buffer(),
```

Comment:
Why we can't use the template function here?

Reply:
done